### DISCUSSION TITLE
How to implement a new Datasource integration?

### DISCUSSION THREAD

**Discussion by axoxymous-coder** [Canner/WrenAI#982] [Q&A]
Hi Team,

I wanted to use WrenAI with Amazon Redshift. However, since Redshift is not yet supported by WrenAI, thought of implementing  the support on my own. Is there a document or a HowTo which I can refer to for implementation  of a new datasource integration? I have looked at the Ibis Server's code where individual datasources are implemented. Is that the only place where I need to implement the interfaces? Any pointers or help will be highly appreciated.

Thanks.

**Comment by cyyeh** [comment]
@axoxymous-coder Hi, thanks for reaching out! For guide on how to add a data source, please check out this document here: https://github.com/Canner/wren-engine/blob/main/ibis-server/docs/how-to-add-data-source.md. If you have other questions, welcome to discuss with us on Discord: https://discord.gg/5DvshJqG8Z

**Comment by axoxymous-coder** [comment]
Thank you @cyyeh 

I was thinking of implementing  a datasource connector for AWS RedShift. But it seems, Ibis project itself does not support a Redshift backend.
